GitTfs/Commands/Fetch.cs

```diff
@@ -146,6 +146,9 @@ private void FetchRemote(bool stopOnFailMergeCommit, IGitTfsRemote remote)
 
         protected virtual void DoFetch(IGitTfsRemote remote, bool stopOnFailMergeCommit)
         {
+            if (upToChangeSet > 0 && InitialChangeset.HasValue && InitialChangeset.Value > upToChangeSet)
```

Comment:
`-1` is a pretty common way to indicate lack of something (index of a non-existent array element, range without upper bound, etc.) so I don't think it's [really] necessary to replace it with a nullable value or anything like that.
It probably would be easier/better to use `Int32.MaxValue` as the *internal* representation of "all changesets up to the latest one" — extremely low probability of finding a repository with 2 billion changesets means that `upToChangeSet == Int32.MaxValue` is pretty much the same thing as the current `upToChangeSet == -1`, but can be used without any special checks.